packages/evm/src/interpreter.ts

```diff
@@ -374,6 +379,13 @@ export class Interpreter {
 
     let gas = opInfo.feeBigInt
 
+    // Cache pre-gas memory size if doing tracing (EIP-3155)
+    let memorySize = BIGINT_0
+
+    if (this._evm.events.listenerCount('step') > 0 || this._evm.DEBUG) {
+      memorySize = this._runState.memoryWordCount
```

Comment:
I would skip this entire if statement and just cache the memory size, I think that's faster than doing the actual check